dbms/include/DB/DataStreams/ColumnGathererStream.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <DB/DataStreams/IProfilingBlockInputStream.h>
+#include <DB/Storages/IStorage.h>
+#include <DB/Common/PODArray.h>
+
+
+namespace DB
+{
+
+
+struct __attribute__((__packed__)) RowSourcePart
+{
+	unsigned int skip: 1;
```

Comment:
It is better to have getFlag() getSourceNum() methods instead of using of explicit bit ops in calling code, so raw UInt8 alias is not convinient.
Bit fields can be used instead of such methods.